### PR TITLE
move [R] warnings out of destructor path

### PR DESCRIPTION
**Pull request by detule**
Hi Team:

Just following up on a discussion where @hadley pointed out we should avoid `Rcpp::warning` in the destructor path as an error there, as a result of `options(warn>2)`, and the subsequent `longjmp`, could lead to incorrect resource unwind.

Moving the warning up to the [R] frame.

### CODE REVIEW

File: R/dbi-connection.R [hunk 1]
```diff
@@ -81,6 +81,10 @@ setMethod("dbDisconnect", "OdbcConnection",
 #' @export
 setMethod("dbSendQuery", c("OdbcConnection", "character"),
   function(conn, statement, params = NULL, ..., immediate = FALSE) {
+    if (has_result(conn@ptr))
+    {
```
[the remark this answers is not in the document]
Comment (simonpcouch) [reply R/dbi-connection.R:85]: Just two changes re: tidyverse style guide here.

[the remark this answers is not in the document]
Comment (detule) [reply R/dbi-connection.R]: Done / thanks

File: R/dbi-connection.R [hunk 2]
```diff
@@ -109,6 +113,10 @@ setMethod("dbExecute", c("OdbcConnection", "character"),
 #' @export
 setMethod("dbSendStatement", c("OdbcConnection", "character"),
   function(conn, statement, params = NULL, ..., immediate = FALSE) {
+    if (has_result(conn@ptr))
+    {
```
[the remark this answers is not in the document]
Comment (detule) [reply R/dbi-connection.R]: Done / thanks

File: R/dbi-connection.R [hunk 3]
```diff
@@ -81,6 +81,10 @@ setMethod("dbDisconnect", "OdbcConnection",
 #' @export
 setMethod("dbSendQuery", c("OdbcConnection", "character"),
   function(conn, statement, params = NULL, ..., immediate = FALSE) {
+    if (has_result(conn@ptr))
+    {
+      warning("Cancelling previous query")
```
[the remark this answers is not in the document]
Comment (detule) [reply R/dbi-connection.R]: Done / thanks

File: R/dbi-connection.R [hunk 4]
```diff
@@ -109,6 +113,10 @@ setMethod("dbExecute", c("OdbcConnection", "character"),
 #' @export
 setMethod("dbSendStatement", c("OdbcConnection", "character"),
   function(conn, statement, params = NULL, ..., immediate = FALSE) {
+    if (has_result(conn@ptr))
+    {
+      warning("Cancelling previous query")
```
[the remark this answers is not in the document]
Comment (detule) [reply R/dbi-connection.R]: Done / thanks

**Review by hadley — APPROVED** [review]

**Pull request merged**